vmdb/app/models/customization_script.rb

```diff
@@ -1,4 +1,7 @@
 class CustomizationScript < ActiveRecord::Base
   include NewWithTypeStiMixin
   belongs_to :provisioning_manager
+
+  scope :ptables, where(:type => "CustomizationScriptPtable")
+  scope :media, where(:type => "CustomizationScriptMedium")
```

Comment:
Sorry, yeah, I confused myself entirely. They are indeed class methods.

The one feature is that your lambda can return nil, and it'll do the right thing. I wouldn't consider that enough of a reason to create such a concept, but as it's there... to me, it helps distinguish the simple named filters from the possibly-numerous other class methods. Certainly once your scope starts taking parameters, then it's almost always going to be more confusing than helpful.